cmake/configure/configure_40_cuda.cmake

```diff
@@ -61,6 +61,16 @@ macro(feature_cuda_find_external var)
         )
       endif()
 
+      #
+      # We do not support CUDA 12.0 and newer:
+      #
+      if(CUDA_VERSION VERSION_GREATER_EQUAL 12.0)
+        message(FATAL_ERROR "\n"
+          "deal.II's own CUDA backend does not support CUDA version 12.0 or newer.\n"
+          "Instead, configure Kokkos with CUDA enabled."
+        )
+      endif()
```

Comment:
I think this is reasonable anyway (independent of our decision to fix it eventually or not).
I am not quite sure about deprecating, though, until we have some better replacement for matrix-based GPU support (PETSc or Trilinos+Tpetra).
Also, I'm not entirely sure that we pull in all `Cuda` libraries when compiling deal.II with `Kokkos+Cuda` but without `DEAL_II_WITH_CUDA=ON` (at least before https://github.com/dealii/dealii/pull/15283).